src/snapred/ui/workflow/WorkflowImplementer.py

```diff
@@ -91,6 +96,9 @@ def reset(self, retainOutputs=False):
         self.outputs = []
         self.collectedOutputs = []
 
+        for hook in self.resetHooks:
```

Comment:
See also, all of the `startLambda, resetLambda, iterateLambda, ...` at `Workflow.py`.  However, I think your approach of using (possibly multiple) _hook_ functions is probably _better_.  (And, that naming messes with the actual meaning of `lambda` as being _anonymous_.)

Reply:
Yeah, Michael had mentioned this but I figured this way would be a bit more generalized.